File: item-list/src/main/java/com/gurunars/item_list/Change.java
package com.gurunars.item_list;

import android.support.v7.widget.RecyclerView;

import java.util.List;

interface Change<ItemType extends Item> {
    int apply(
            RecyclerView.Adapter adapter,
            Scroller scroller,
            List<ItemType> items,
            int currentPosition);
}

File: item-list/src/main/java/com/gurunars/item_list/Change.kt
package com.gurunars.item_list

import android.support.v7.widget.RecyclerView

internal interface Change<ItemType : Item> {
    fun apply(
            adapter: RecyclerView.Adapter<*>,
            scroller: Scroller,
            items: MutableList<ItemType>,
            currentPosition: Int): Int
}
